scripts/staticx_patch.diff

```diff
@@ -0,0 +1,28 @@
+diff --git a/libtar/decode.c b/libtar/decode.c
+index b34b1d9..d221c3a 100644
+--- a/libtar/decode.c
++++ b/libtar/decode.c
+@@ -44,11 +44,6 @@ uid_t
+ th_get_uid(const TAR *t)
+ {
+ 	int uid;
+-	struct passwd *pw;
+-
+-	pw = getpwnam(t->th_buf.uname);
+-	if (pw != NULL)
+-		return pw->pw_uid;
+ 
+ 	/* if the password entry doesn't exist */
+ 	sscanf(t->th_buf.uid, "%o", &uid);
+@@ -60,11 +55,6 @@ gid_t
+ th_get_gid(const TAR *t)
+ {
+ 	int gid;
+-	struct group *gr;
+-
+-	gr = getgrnam(t->th_buf.gname);
+-	if (gr != NULL)
+-		return gr->gr_gid;
+ 
+ 	/* if the group entry doesn't exist */
+ 	sscanf(t->th_buf.gid, "%o", &gid);
```

Comment:
@LiorMosk do you mind checking. I actually checked the unpacked files of staticx and they look alright. But I still want to verify that the contents of `t->th_buf.gid` (and same for `t->th_buf.uid`) are always legit in our case (i.e no chance that if `uname` is set to `"root"` then `uid` is not set because `root` is expected to always be processes by `getpwnam`, for example...)

Read the sources or maybe build staticx with prints and see?

Reply:
It seems to be ok, that way